sriovnet.go

```diff
@@ -450,15 +462,55 @@ func GetNetDevicesFromPci(pciAddress string) ([]string, error) {
 	return netDevices, nil
 }
 
-// GetPfPciFromVfPci retrieves the parent PF PCI address of the provided VF PCI address in D:B:D.f format
-func GetPfPciFromVfPci(vfPciAddress string) (string, error) {
-	pfPath := filepath.Join(PciSysDir, vfPciAddress, "physfn")
-	pciDevDir, err := utilfs.Fs.Readlink(pfPath)
+// GetNetDeviceFromAux gets auxiliary device name (e.g 'mlx5_core.sf.2') and
```

Comment:
> also i wonder (in general) if we could leverage devlink instead of parsing sysfs for SFs
> 
> i.e devlink port

Then there will be two dependancies: on devlink and on mlxdevm

Reply:
which is fine IMO, 

devlink for upstream
mlxdevm for old kernel + MOFED

we should use these interfaces and fallback to sysfs if not supported. 
later on the sysfs bits can be removed.

@DmytroLinkin do you think its a large effort to support ? i think its the correct approach for SFs

Reply:
I personally prefer to keep everithing simple and don't rely on any binaries. Ideally to get list of SFs rtnetlink shoud be used, but this approach doesn't fit into "simple" :D

As for devlink/mlxdevm/sysfs. In SF case sysfs can be dropped because SFs can't be created without devlink or mlxdevm, but that's not a case for other aux devices.
It is not a big effort to impement interface, but it should be used for VFs also.